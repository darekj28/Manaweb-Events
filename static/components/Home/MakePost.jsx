var React = require('react');
import FilterButton from "./FilterButton.jsx";

export default class MakePost extends React.Component {
	componentDidMount() {
		$('.post-button').click(function() {
			$(this).blur();
		});
		$('#MessagePost').hide();
		$('#TogglePost').click(function() {
			$('#MessagePost').slideToggle(function() {
				$('#PostInput').focus();
			});
		})
	}
	handlePostSubmit() {
		if (this.postText.value.trim().length > 0)
			this.props.onPostSubmit(this.postText.value);
	}
	handlePostChange() {
		this.props.onPostChange(this.postText.value);
	}
	render() {
		return(
			<div> 
				<div id="TogglePost"><h4>Post a message...</h4></div>
				<div id="MessagePost">
					<textarea id="PostInput" className="PostText form-control" 
							value={this.props.postText} rows="4"
							placeholder={this.props.placeholder} ref={(input) => this.postText = input} 
							onSubmit={this.handlePostSubmit} onChange={this.handlePostChange.bind(this)}></textarea>
					{this.props.actions.map(function(action, i) {
						return <FilterButton key={i} onClick={this.props.onClick} selected={false} name={action}/>;
					}, this)}
					<div className="input-group-addon"
							onClick={this.handlePostSubmit.bind(this)}>
						<button className="btn post-button">POST!</button>
					</div>	
				</div>
			</div>);
	}
}