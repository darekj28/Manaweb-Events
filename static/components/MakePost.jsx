var React = require('react');
// var $ = require('jquery');
import FilterButton from "./FilterButton.jsx";

export default class MakePost extends React.Component {
	constructor(props) {
		super(props);
		this.handlePostChange = this.handlePostChange.bind(this);
		this.handlePostSubmit = this.handlePostSubmit.bind(this);
		this.handleEnterPress = this.handleEnterPress.bind(this);
	}
	componentDidMount() {
		var messageVisible = true;
		var that = this;

		$('#SubmitButtonPost').click(function() {
			$(this).blur();
		});
		$('#StartButtonPost').click(function(){
			$(this).blur();
	        if(!messageVisible){
	            messageVisible = !messageVisible;
	            $(this).siblings('#MessagePost').fadeIn(function(){
	                $(this).children('#PostTextPost').focus();
	            });
	        } else {
	            messageVisible = !messageVisible;
	            $(this).siblings('#MessagePost').fadeOut(function(){
	                $(this).children('#PostTextPost').blur();
	            });
	        } 
	    });
	    
	}
	handlePostSubmit() {
		if (this.postText.value.length > 0)
			this.props.onPostSubmit(this.postText.value);
	}
	handlePostChange() {
		this.props.onPostChange(this.postText.value);
	}
	handleEnterPress(target) {
		if (this.postText.value.length > 0)
		    if (target.charCode==13)
		        this.props.onPostSubmit(this.postText.value); 
	}
	render() {
		var that = this;
		return(
			<div> 
				<a id="StartButtonPost" className="StartButton">
					<span className="glyphicon glyphicon-pencil AppGlyphicon"></span>
				</a>
				<div id="MessagePost" className="Message pull-right input-group input-group-unstyled">
					<input id="PostTextPost" type="text" className="PostText form-control" 
							onKeyPress={this.handleEnterPress} value={this.props.postText} 
							placeholder={this.props.placeholder} ref={(input) => this.postText = input} 
							onSubmit={this.handlePostSubmit} onChange={this.handlePostChange}></input>
					{this.props.actions.map(function(action, i) {
						if (action == "Trade") return <FilterButton key={i} onClick={that.props.onClick} active={true} name={action}/>;
						else return <FilterButton key={i} onClick={that.props.onClick} active={false} name={action}/>;
					})}
					<a className="SubmitButton input-group-addon" id="SubmitButtonPost" 
							onClick={this.handlePostSubmit}>
						<span className="glyphicon glyphicon-send AppGlyphicon"></span>
					</a>
				
				</div>
			</div>);
	}
}