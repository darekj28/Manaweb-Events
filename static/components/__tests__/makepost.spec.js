import MakePost from "../MakePost.jsx";

describe("MakePost", ()=>{
	it("renders correctly", () => {
		const actions = ["Trade", "Play", "Chill"];
		const mp = renderer.create(
			<MakePost postText="a" onClick={jest.fn()} onPostChange={jest.fn()}
				onPostSubmit={jest.fn()} actions={actions}/>
			);
		expect(mp).toMatchSnapshot();
	});
	it("should call handlePostChange with text in textbox when user types", () => {
		expect(true).toBe(true);
	});
	it("should call handlePostSubmit with text in textbox when user presses submit or enter", () => {
		expect(true).toBe(true);
	});
});