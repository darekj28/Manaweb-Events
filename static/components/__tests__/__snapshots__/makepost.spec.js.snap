exports[`MakePost renders correctly 1`] = `
<div>
  <a
    className="StartButton"
    id="StartButtonPost">
    <span
      className="glyphicon glyphicon-pencil AppGlyphicon" />
  </a>
  <div
    className="Message pull-right input-group input-group-unstyled"
    id="MessagePost">
    <input
      className="PostText form-control"
      id="PostTextPost"
      onChange={[Function]}
      onKeyPress={[Function]}
      onSubmit={[Function]}
      placeholder={undefined}
      type="text"
      value="a" />
    <a
      className="input-group-addon">
      <span
        className="glyphicon glyphicon-transfer icon-success filterButton"
        onClick={[Function]} />
    </a>
    <a
      className="input-group-addon">
      <span
        className="glyphicon glyphicon-play icon-danger filterButton"
        onClick={[Function]} />
    </a>
    <a
      className="input-group-addon">
      <span
        className="glyphicon glyphicon-time icon-danger filterButton"
        onClick={[Function]} />
    </a>
    <a
      className="SubmitButton input-group-addon"
      id="SubmitButtonPost"
      onClick={[Function]}>
      <span
        className="glyphicon glyphicon-send AppGlyphicon" />
    </a>
  </div>
</div>
`;
